src/SoundTestIC.cpp

```diff
@@ -0,0 +1,33 @@
+#include "SoundTestIC.h"
+#include "ComponentsManager.h"
+#include "Entity.h"
+#include "FactoriesFactory.h"
+#include "Scene.h"
+
+void SoundTestIC::handleInput(const SDL_Event& _event) {
+
+    if (soundComponent_ == nullptr)
+        soundComponent_ = dynamic_cast<SoundComponent*>(
```

Comment:
Avoid RTTI, this might need a re-format.

```suggestion
        soundComponent_ = reinterpret_cast<SoundComponent*>(
```